packages/jest-worker/src/Farm.ts

```diff
@@ -44,41 +46,60 @@ export default class Farm {
     }
   }
 
-  doWork(method: string, ...args: Array<any>): Promise<unknown> {
-    return new Promise((resolve, reject) => {
-      const computeWorkerKey = this._computeWorkerKey;
-      const request: ChildMessage = [CHILD_MESSAGE_CALL, false, method, args];
+  doWork(
+    method: string,
+    ...args: Array<any>
+  ): PromiseWithCustomMessage<unknown> {
+    let customMessageListeners: Array<OnCustomMessage> = [];
```

Comment:
should we make this into a `Set` instead?

Reply:
Yeah, that makes sense, it would avoid duplicate message entries too and provide more clarity. Also easier to clear the set (better than undefined in the arrays xD). I've added the update